tensorflow/compiler/xla/service/gpu/horizontal_loop_fusion.cc

```diff
@@ -189,14 +215,17 @@ bool IsProfitableFusionCandidate(const HloInstruction& instr) {
 // The horizontal fusion excludes computations with non-row-major layouts,
 // because fusing computations with different layouts can result in uncoalesced
 // memory accesses and cause great performance overhead.
-bool HasOnlyRowMajorLayout(const HloInstruction& fusion_instr) {
-  CHECK(fusion_instr.opcode() == HloOpcode::kFusion);
-  auto instrs = fusion_instr.fused_instructions_computation()->instructions();
-  for (auto instr : instrs) {
-    if (instr->shape().layout().format() != DENSE) {
+bool HasOnlyRowMajorLayout(const HloInstruction& instr) {
+  if (instr.opcode() != HloOpcode::kFusion) {
+    return LayoutUtil::IsMonotonicWithDim0Major(instr.shape().layout());
```

Comment:
If it is not fusion, it is elementwise here. My understanding is that operands of the elementwise ops are `DENSE` and so we don't need to check whether they are `DENSE` here and can go directly to invoke `IsMonotonicWithDim0Major()`.

Let me know if this is not the case.